src/components/PatronForm/PatronForm.test.js

```diff
@@ -0,0 +1,195 @@
+import {
+  fireEvent,
+  render,
+  screen,
+} from '@testing-library/react';
+
+import '../../../test/jest/__mock__';
+
+import PatronForm, {
+  getIdentifier,
+} from './PatronForm';
+
+jest.mock('react-final-form', () => ({
+  ...jest.requireActual('react-final-form'),
+  Field: jest.fn(({ onChange, ...rest }) => <div onClick={onChange} {...rest} />),
+}));
+jest.mock('@folio/stripes/final-form', () => () => jest.fn((component) => component));
+
+const testIds = {
+  patronForm: 'patronForm',
+  patronIdentifier: 'patronIdentifier',
+  clickableFindPatronButton: 'clickableFindPatronButton',
+  clickableFindPatronPluggable: 'clickableFindPatronPluggable',
+};
+
+describe('PatronForm', () => {
+  const mockedHandleSubmit = jest.fn();
+  const mockedInitialValues = {};
+  const mockedForm = {
+    change: jest.fn(),
+    reset: jest.fn(),
+    getState: jest.fn(() => ({ values: mockedInitialValues })),
+  };
+  const props = {
+    handleSubmit: mockedHandleSubmit,
+    userIdentifiers: [],
+    submitting: false,
+    submitFailed: false,
+    patron: {},
+    forwardedRef: {},
+    formRef: {},
+    form: mockedForm,
+  };
+  describe('without error', () => {
+    beforeEach(() => {
+      render(
+        <PatronForm
+          {...props}
+        />
+      );
+    });
+
+    afterEach(() => {
+      mockedHandleSubmit.mockClear();
+    });
+
+    it('should render patron form', () => {
+      expect(screen.getByTestId(testIds.patronForm)).toBeInTheDocument();
+    });
+
+    it('should render patron identifier', () => {
+      expect(screen.getByTestId(testIds.patronIdentifier)).toBeInTheDocument();
+    });
+
+    it('should render clickable find patron button', () => {
+      expect(screen.getByTestId(testIds.clickableFindPatronButton)).toBeInTheDocument();
+    });
+
+    it('should render clickable find patron pluggable', () => {
+      expect(screen.getByTestId(testIds.clickableFindPatronPluggable)).toBeInTheDocument();
+    });
+
+    it('should call "handleSubmit" on form submit', () => {
+      expect(mockedHandleSubmit).not.toHaveBeenCalled();
```

Comment:
This `expect` statement is redundant.
On line 54 you have `mockedHandleSubmit.mockClear();` which means that it clears all information stored in the [mockFn.mock.calls](https://jestjs.io/ru/docs/mock-function-api#mockfnmockcalls). 
In other words before each test `mockedHandleSubmit` has 0 number of execution.

Reply:
Fixed